packages/mobile/e2e/scripts/fund-e2e-accounts.ts

```diff
@@ -1,87 +1,173 @@
-dotenv.config({ path: '../.env' })
-
-// Constants
-const valoraE2ETestWallet = '0x6131a6d616a4be3737b38988847270a64bc10caa'
-const valoraTestFaucet = '0xe5F5363e31351C38ac82DBAdeaD91Fd5a7B08846'
-const valoraTestFaucetSecret = process.env.TEST_FAUCET_SECRET
-
-// Init a new kit, connected to the alfajores testnet
-const web3 = new Web3('https://alfajores-forno.celo-testnet.org')
-const kit = ContractKit.newKitFromWeb3(web3)
-
-// Throw error if balance is low
-const balanceError = async (address = valoraE2ETestWallet, minBalance = 10) => {
-  let balanceObject = await getBalance(address)
-  for (const balance in balanceObject) {
-    if (balanceObject[balance] < minBalance) {
-      throw new Error(
-        `${balance} balance of ${address} is below ${minBalance}. Please refill from the faucet https://celo.org/developers/faucet 🙏`
-      )
-    }
-  }
-}
-
-// Default e2e address to lookup
-const getBalance = async (address = valoraE2ETestWallet) => {
-  try {
-    let balanceObj = {}
-
-    // Get Balances
-    let balances = await kit.celoTokens.balancesOf(address)
-
-    // Convert and add to balance object
-    for (const value in balances) {
-      balanceObj[`${value}`] = balances[`${value}`] / 10 ** 18
-    }
-
-    // Return balance object
-    return balanceObj
-  } catch (err) {
-    console.log(err)
-  }
-}
+import { E2E_TEST_FAUCET, E2E_TEST_WALLET } from './consts'
+import { balanceError, dotenv, getBalance, kit, web3 } from './utils'
+dotenv.config({ path: `${__dirname}/../.env` })
 
+const valoraTestFaucetSecret = process.env['TEST_FAUCET_SECRET']!
 ;(async () => {
   // Get E2E Test Wallet Balance & Valora Faucet Balance
   const receivingBalance = await getBalance()
-  const sendingBalance = await getBalance(valoraTestFaucet)
+  const sendingBalance = (await getBalance(E2E_TEST_FAUCET)) ?? {}
+  console.table(await getBalance(E2E_TEST_FAUCET))
 
   // Connect Valora E2E Test Faucet - Private Key Stored in GitHub Secrets
   kit.connection.addAccount(
-    web3.eth.accounts.privateKeyToAccount(valoraTestFaucetSecret).privateKey
+    web3.eth.accounts.privateKeyToAccount(valoraTestFaucetSecret.toString()).privateKey
   )
 
   // Get Token Contract Wrappers
-  let goldtoken = await kit.contracts.getGoldToken()
-  let cUSDtoken = await kit.contracts.getStableToken()
-  let cEURtoken = await kit.contracts.getStableToken('cEUR')
+  const celoToken = await kit.contracts.getGoldToken()
+  const cusdToken = await kit.contracts.getStableToken()
+  const ceurToken = await kit.contracts.getStableToken('cEUR')
+  const celoExchange = await kit.contracts.getExchange()
+  const cusdExchange = await kit.contracts.getExchange('cUSD')
+  const ceurExchange = await kit.contracts.getExchange('cEUR')
+
+  // Balance Faucet
+  let balances = await getBalance(E2E_TEST_FAUCET)
+  let sum = 0
+  const numOfTokens = 3 // Object.keys(balances).length
+  for (let balance in balances) {
+    console.log(balances[balance])
+    sum += balances[balance]
+  }
```

Comment:
Looks like we should sum the specific tokens we are interested in instead, right?

Reply:
Fixed in https://github.com/valora-inc/wallet/pull/2325/commits/b8602f07b02fc7f938f88b39f54eb65550193b59!